test/DeviceTest.java

```diff
@@ -1,82 +1,128 @@
+import org.junit.Assert;
 import org.junit.Test;
-import org.mockito.Mockito;
-import org.mockito.invocation.Invocation;
 
+import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
+import java.lang.reflect.Method;
 import java.util.ArrayList;
-import java.util.Collection;
 
 import static junit.framework.TestCase.fail;
 import static org.mockito.Mockito.spy;
 
 public class DeviceTest {
     @Test
     public void linearTest() {
+        Constructor polynomial = null;
```

Comment:
I added logic that uses reflection to access the private constructors in `TestingDevice.java`.
This is because I wanted TestingDevice.java to match closely to Jody's API in his generated JavaDoc. For that, I made additional fields and constructors not found in his API private, and use reflection to access them in our tests.